### ISSUE TITLE
Error! An error occurred.

### ISSUE THREAD

**Issue by Serialc0de** [shuup/shuup#2334]
![555](https://user-images.githubusercontent.com/40340347/102722124-4c5a1d80-42ff-11eb-9b3a-f5ceb7527af7.PNG)

why?

**Comment by sameerdhoot** [comment]
I also have the same issue. Did not get a chance to troubleshoot but does anyone know of resolution to this?

**Comment by ruohola** [comment]
What is your `DEFAULT_FILE_STORAGE` in `settings.py`?

**Comment by donutnick** [comment]
I also have the same issue.

**Comment by alhague** [comment]
I tried to set up a test instance using docker file (BTW: this runs as root in the default config, seriously?!). I could not get lists of categories under categories and products in admin mode with same error message.

Environment says:
DEFAULT_FILE_STORAGE | 'django.core.files.storage.FileSystemStorage'

Traceback reads:

File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/usr/local/lib/python3.7/dist-packages/shuup/admin/utils/urls.py" in _wrapped_view
  103.             return view_func(request, *args, **kwargs)

File "/usr/local/lib/python3.7/dist-packages/django/views/generic/base.py" in view
  71.             return self.dispatch(request, *args, **kwargs)

File "/usr/local/lib/python3.7/dist-packages/django/views/generic/base.py" in dispatch
  97.         return handler(request, *args, **kwargs)

File "/usr/local/lib/python3.7/dist-packages/shuup/admin/utils/picotable.py" in get
  503.             return self.process_picotable(query)

File "/usr/local/lib/python3.7/dist-packages/shuup/admin/utils/picotable.py" in process_picotable
  498.         return JsonResponse(pico.get_data(json.loads(query_json)), encoder=ExtendedJSONEncoder)

File "/usr/local/lib/python3.7/dist-packages/shuup/admin/utils/picotable.py" in get_data
  447.             "items": [self.process_item(item) for item in page],

File "/usr/local/lib/python3.7/dist-packages/shuup/admin/utils/picotable.py" in <listcomp>
  447.             "items": [self.process_item(item) for item in page],

File "/usr/local/lib/python3.7/dist-packages/shuup/admin/modules/products/views/list.py" in process_item
  27.         out = super(ProductPicotable, self).process_item(object)

File "/usr/local/lib/python3.7/dist-packages/shuup/admin/utils/picotable.py" in process_item
  466.             out[column.id] = column.get_display_value(context=self.context, object=object)

File "/usr/local/lib/python3.7/dist-packages/shuup/admin/utils/picotable.py" in get_display_value
  324.             return display_callable(object)

File "/usr/local/lib/python3.7/dist-packages/shuup/admin/modules/products/views/list.py" in get_primary_image
  148.             thumbnail = instance.product.primary_image.get_thumbnail()

File "/usr/local/lib/python3.7/dist-packages/shuup/core/models/_product_media.py" in get_thumbnail
  151.             return thumbnailer.get_thumbnail(thumbnail_options=kwargs)

File "/usr/local/lib/python3.7/dist-packages/easy_thumbnails/files.py" in get_thumbnail
  516.                     silent_template_exception=silent_template_exception)

File "/usr/local/lib/python3.7/dist-packages/easy_thumbnails/files.py" in generate_thumbnail
  390.                                                self.thumbnail_processors)

File "/usr/local/lib/python3.7/dist-packages/easy_thumbnails/engine.py" in process_image
  34.         image = processor(image, **processor_options)

File "/usr/local/lib/python3.7/dist-packages/easy_thumbnails/processors.py" in scale_and_crop
  199.                        resample=Image.ANTIALIAS)

File "/usr/lib/python3/dist-packages/PIL/Image.py" in resize
  1804.         self.load()

File "/usr/lib/python3/dist-packages/PIL/ImageFile.py" in load
  139.         pixel = Image.Image.load(self)

File "/usr/lib/python3/dist-packages/PIL/Image.py" in load
  859.             return self.im.pixel_access(self.readonly)

File "/usr/lib/python3/dist-packages/PIL/_util.py" in __getattr__
  30.         raise self.ex

Exception Type: ValueError at /sa/products/
Exception Value: Operation on closed image

digging deeper it seems there was an issue with an image, so I did remove *.jpeg from filer_public. This let me see the lists but any upload of a product image broke it again because thumbnail creation fails. File upload itself works, a file in generated under var/media/filer_public 

I then saw that in models/_product_media.py that only InvalidImageFormatError is cought, so everything breaks when a different error occurrs. So quickfix for me was to change 
/usr/local/lib/python3.7/dist-packages/shuup/core/models/_product_media.py:152 to except Exception: which leaves me with no thumbnails but at least I'm not totally blocked from further evaluation.

**Comment by Bobby00** [comment]
I got the same error. What I noticed is when you fork/clone the repo and run the quick "docker-compose up", it only builds a mini version of the Shuup platform, mostly for quick testing purposes. I got it to work by setting it up manually, there are two options:-

1. Reconfigure existing Docker files to install all dependencies and run the command to build the Nodejs resource that is listed in this link below :
 
python setup.py build_resources
https://shuup.readthedocs.io/en/latest/howto/getting_started_dev.html


2. Or setup the whole project manually locally using instructions from above link

**Comment by ruohola** [comment]
@Bobby00 Yes, the `docker-compose.yml` file is for just getting Shuup running quickly it's not made for development. The `docker-compose-dev.yml` file can be developed with. You can use it with:

```
$ docker-compose -f docker-compose-dev.yml up
```

**Comment by Serialc0de** [comment]
too difficult to install this subspecies of e-commerce. Change the installation is too complex, too many problems ... only professional developers can install it, then apart from everything it really sucks me!

**Comment by martinlamas** [comment]
I also had the same issue. I'm running the application in a docker environment. Changing Python version 3.7 to 3.6 solved it for me.

**Comment by thusinh1969** [comment]
This is TOO complicated even to get a demo up and running. I gave up !

**Comment by Nzamuyejoram** [comment]
@thusinh1969 there is no giving up plz on the issue. lets look on the best way to troubleshoot
